### PR TITLE
Added following tests to index under 'test/misc'.

### PR DESCRIPTION
**Pull request by girish946**
* alias_alloc.lua
* fori_coerce.lua
* gc_rechain.lua
* gcstep.lua
* gc_trace.lua
* hook_norecord.lua
* parse_comp.lua

### CODE REVIEW

File: test/misc/index [hunk 1]
```diff
@@ -1 +1,7 @@
+alias_alloc.lua
 debug.lua
+gc_trace.lua
+gc_rechain.lua
+gcstep.lua
+hook_norecord.lua
+fori_coerce.lua
```
Comment (siddhesh) [inline test/misc/index]: Add this in alphabetical order.

File: test/misc/parse_comp.lua [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
   assert(1 < a)
   assert(1 < (f[1].n))
   assert(1 < f[1].n)
 end
 
-do
+do --- smoke
   tt = { a = 1 }
   assert(not(0 >= tt.a))
 end
```
Comment (siddhesh) [inline test/misc/parse_comp.lua]: Add `parse_comp.lua` in index.

**Review by siddhesh — CHANGES_REQUESTED** [review]
Almost there, please fix up the issues pointed out in the review comments and you're good to go.  Thanks @girish946 !

File: test/misc/index (hunk 1 above)
Comment (girish946) [reply test/misc/index]: done.

File: test/misc/parse_comp.lua (hunk 2 above)
Comment (girish946) [reply test/misc/parse_comp.lua]: done.

**Review by siddhesh — APPROVED** [review]

**Pull request merged**